src/renderer/routes/test_sequencer_panel/components/DataTable.tsx

```diff
@@ -0,0 +1,525 @@
+import {
+  ContextMenu,
+  ContextMenuContent,
+  ContextMenuItem,
+  ContextMenuTrigger,
+} from "@/renderer/components/ui/context-menu";
+import * as React from "react";
+import {
+  ColumnDef,
+  ColumnFiltersState,
+  Row,
+  SortingState,
+  VisibilityState,
+  flexRender,
+  getCoreRowModel,
+  getFilteredRowModel,
+  getPaginationRowModel,
+  getSortedRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+
+import { Button } from "@/renderer/components/ui/button";
+import { Checkbox } from "@/renderer/components/ui/checkbox";
+import {
+  DropdownMenu,
+  DropdownMenuCheckboxItem,
+  DropdownMenuContent,
+  DropdownMenuTrigger,
+} from "@/renderer/components/ui/dropdown-menu";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/renderer/components/ui/table";
+import {
+  TestSequenceElement,
+  ConditionalComponent,
+  Conditional,
+  Test,
+  StatusTypes,
+} from "@/renderer/types/testSequencer";
+import { useTestSequencerState } from "@/renderer/hooks/useTestSequencerState";
+import { parseInt, filter, map } from "lodash";
+import { AddConditionalModal } from "./AddConditionalModal";
+import {
+  generateConditional,
+  getIndentLevels,
+} from "../utils/ConditionalUtils";
+import {
+  ChevronUpIcon,
+  ChevronDownIcon,
+  Loader,
+  TrashIcon,
+} from "lucide-react";
+import { WriteConditionalModal } from "./AddWriteConditionalModal";
+import LockableButton from "./lockable/LockedButtons";
+
+const IndentLine = ({
+  name,
+  level = 0,
+}: {
+  name: React.ReactNode;
+  level: number;
+}) =>
+  level == 0 ? (
+    <div className="relative ml-5 pl-1">
+      <div style={{ marginLeft: level == 0 ? `${level * 20}px` : 0 }}>
+        {level == 0 ? name : <IndentLine name={name} level={level - 1} />}
+      </div>
+    </div>
+  ) : (
+    <div className="relative ml-5 pl-1">
+      <div
+        className={"border-l-2 border-blue-800 py-1 pl-4"}
+        style={{ marginLeft: level == 0 ? `${level * 20}px` : 0 }}
+      >
+        {level == 0 ? name : <IndentLine name={name} level={level - 1} />}
+      </div>
+    </div>
```

Comment:
you can remove the outer condition by just adding a conditional style on the inner div